File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        set<int>oset;
        for(int i=0;i<nums.size();i++)
        {
            if(oset.size()<k)
            {
                if(oset.count(nums[i]))return true;
                oset.insert(nums[i]);
            }
            else
            {
                if(oset.count(nums[i]))return true;
                oset.insert(nums[i]);
                oset.erase(nums[i-k]);
            }
        }
        return false;
    }
};

File: 0219-contains-duplicate-ii/0219-contains-duplicate-ii.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        Map<Integer,Integer> mp = new HashMap<>();
        for(int i=0;i<nums.length;i++)
        {
            if(!mp.containsKey(nums[i]))
            {
                mp.put(nums[i],i);
                continue;
            }
            if(i-mp.get(nums[i])<=k)
                return true;
            mp.put(nums[i],i);
        }

        return false;
    }
}